guides/source/testing.md

```diff
@@ -8,72 +8,90 @@ This guide covers built-in mechanisms in Rails for testing your application.
 After reading this guide, you will know:
 
 * Rails testing terminology.
-* How to write unit, functional, integration, and system tests for your application.
+* How to write unit, functional, integration, and system tests for your
+  application.
 * Other popular testing approaches and plugins.
 
 --------------------------------------------------------------------------------
 
-Why Write Tests for Your Rails Applications?
---------------------------------------------
+Why Write Tests?
+----------------
 
-Rails makes it super easy to write your tests. It starts by producing skeleton test code while you are creating your models and controllers.
+Rails makes it super easy to write your tests. It starts by producing skeleton
+test code when you create your models and controllers.
 
-By running your Rails tests you can ensure your code adheres to the desired functionality even after some major code refactoring.
+By running your Rails tests you can quickly and easily ensure your code still
+adheres to the desired functionality even after some major code refactoring.
```

Comment:
Something like this to explain the "why" of testing? From m perspective, it's about discovering bugs early and building confidence. My suggestion but feel free to reword:

```suggestion
By running tests after code changes such as adding new features or refactoring, you can quickly find out if existing code still works as expected. You can discover and fix any issue that are uncovered by failing tests. By running tests, you can discover and fix bugs early in the development cycle and improve confidence.
```

Reply:
Thanks everyone - this was really nice because I've ended up with an amalgamation of everyone's ideas and my own re-write - what a team effort paragraph!